common/src/main/java/org/opensearch/ml/common/indexInsight/FieldDescriptionTask.java

```diff
@@ -0,0 +1,320 @@
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.transport.client.Client;
+
+import com.jayway.jsonpath.JsonPath;
+
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Field Description Task: Generates descriptions for index fields using LLM.
+ * This task analyzes index mapping and sample data to provide meaningful descriptions
+ * for each field in the index, helping down-stream tasks understand the purpose and content of fields.
+ */
+@Log4j2
+public class FieldDescriptionTask implements IndexInsightTask {
+
+    private static final int BATCH_SIZE = 50; // Hard-coded value for now
+    private static final MLIndexInsightType TASK_TYPE = MLIndexInsightType.FIELD_DESCRIPTION;
+    private final String sourceIndex;
+    private final Client client;
+
+    public FieldDescriptionTask(String sourceIndex, Client client) {
+        this.sourceIndex = sourceIndex;
+        this.client = client;
+    }
+
+    @Override
+    public void runTask(String storageIndex, String tenantId, ActionListener<IndexInsight> listener) {
+        try {
+            getInsightContentFromContainer(storageIndex, MLIndexInsightType.STATISTICAL_DATA, ActionListener.wrap(statisticalContent -> {
+                getAgentIdToRun(client, tenantId, ActionListener.wrap(agentId -> {
+                    batchProcessFields(statisticalContent, agentId, storageIndex, listener);
+                }, listener::onFailure));
+            }, e -> {
+                log.error("Failed to get statistical content for index {}", sourceIndex, e);
+                saveFailedStatus(storageIndex);
+                listener.onFailure(e);
+            }));
+        } catch (Exception e) {
+            log.error("Failed to execute field description task for index {}", sourceIndex, e);
+            saveFailedStatus(storageIndex);
+            listener.onFailure(e);
+        }
+    }
+
+    @Override
+    public MLIndexInsightType getTaskType() {
+        return TASK_TYPE;
+    }
+
+    @Override
+    public String getSourceIndex() {
+        return sourceIndex;
+    }
+
+    @Override
+    public Client getClient() {
+        return client;
+    }
+
+    @Override
+    public List<MLIndexInsightType> getPrerequisites() {
+        return Collections.singletonList(MLIndexInsightType.STATISTICAL_DATA);
+    }
+
+    private void batchProcessFields(String statisticalContent, String agentId, String storageIndex, ActionListener<IndexInsight> listener) {
+        Map<String, Object> mappingSource;
+        try {
+            mappingSource = (Map<String, Object>) ((Map<String, Object>) JsonPath.read(statisticalContent, "$")).get("mapping");
```

Comment:
Is it possible the `getInsightContentFromContainer` returns a Map<String, Object>` type and here we can avoid using JsonPath to covert string?